aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -1196,7 +1211,21 @@ public void subscribeToTopic(String topic, AWSIotMqttQos qos,
 
         if (null != mqttClient) {
             try {
-                mqttClient.subscribe(topic, qos.asInt());
+                if (subscriptionStatusCallback != null) {
+                    mqttClient.subscribe(topic, qos.asInt(), null, new IMqttActionListener() {
+                        @Override
+                        public void onSuccess(IMqttToken asyncActionToken) {
+                            subscriptionStatusCallback.onSuccess();
+                        }
+
+                        @Override
+                        public void onFailure(IMqttToken asyncActionToken, Throwable exception) {
+                            subscriptionStatusCallback.onFailure(exception);
+                        }
+                    });
+                } else {
+                    mqttClient.subscribe(topic, qos.asInt());
+                }
             } catch (final MqttException e) {
                 throw new AmazonClientException("Client error when subscribing.", e);
```

Comment:
Consider passing this exception through the callback so that success and failure are all passed via the callback consistently.

Reply:
This handler is for the flavor of subscribe that does not have any callback. For the one with callback, the exception will be passed in the callback itself.